File: src/Entitybus.java
import javax.persistence.*;


/**
 * Created by Olga Pavlova on 9/16/2016.
 */
@Entity
@Table(name = "ENTITYBUS", uniqueConstraints={@UniqueConstraint(columnNames={"ENUMBER"})})//(name = "ENTITYBUS" , schema = "INFORMATION_SCHEMA", catalog = "TEST")
@NamedQuery(name = "My.Test", query = "SELECT e FROM Entitybus e WHERE e.enumber like :name")
@NamedNativeQueries({
        @NamedNativeQuery(
                name = "findStockByStockCodeNativeSQL",
                query = "select * from ENTITYBUS s where s.ENUMBER = :arg",
                resultClass = entity.Entitybus.class
        )
})
@SqlResultSetMapping(
        name = "BusValueMapping",
        classes = @ConstructorResult(
                targetClass = entity.Entitybus.class,
                columns = {
                        @ColumnResult(name = "id", type = Long.class),
                        @ColumnResult(name = "enumber")}))
public class Entitybus
         {
    private int eid;


    @Id
    @SequenceGenerator(name="CUST_SEQ", allocationSize=25)
    @GeneratedValue(strategy=SEQUENCE, generator="CUST_SEQ")
    @Column(name = "EID", nullable = false)
    public int getEid() {
        return eid;
    }

    public void setEid(int eid) {
        this.eid = eid;
    }


    private String enumber;
    @Basic

    @Column(name = "ENUMBER", nullable = true, length = 255)
    public String getEnumber() {
        return enumber;
    }

    public void setEnumber(String enumber) {
        this.enumber = enumber;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        entity.Entitybus entitybus = (entity.Entitybus) o;

        if (eid != entitybus.eid) return false;
        if (enumber != null ? !enumber.equals(entitybus.enumber) : entitybus.enumber != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = eid;
        result = 31 * result + (enumber != null ? enumber.hashCode() : 0);
        return result;
    }
}

File: src/entity/Entitybus.kt
package entity

import javax.persistence.*
import javax.persistence.GenerationType.SEQUENCE

/**
 * Created by Olga Pavlova on 9/16/2016.
 */
@Entity
@Table(name = "ENTITYBUS", uniqueConstraints = arrayOf(UniqueConstraint(columnNames = arrayOf("ENUMBER"))))
@NamedQuery(name = "My.Test", query = "SELECT e FROM Entitybus e WHERE e.enumber like :name")
@NamedNativeQueries(NamedNativeQuery(name = "findStockByStockCodeNativeSQL", query = "select * from ENTITYBUS s where s.ENUMBER = :arg", resultClass = Entitybus::class))
//@SqlResultSetMapping(name = "AuthorValueMapping", classes = ConstructorResult(targetClass = Entitybus::class, columns = arrayOf(ColumnResult(name = "id", type = Long::class), ColumnResult(name = "firstname"), ColumnResult(name = "lastname"), ColumnResult(name = "numBooks", type = Long::class))))
class Entitybus {
    @get:Id
    @SequenceGenerator(name = "CUST_SEQ", allocationSize = 25)
    @GeneratedValue(strategy = SEQUENCE, generator = "CUST_SEQ")
    @get:Column(name = "EID", nullable = false)
    var eid: Int = 0

    @get:Basic
    @get:Column(name = "ENUMBER", nullable = true, length = 255)
    var enumber: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val entitybus = o as Entitybus?

        if (eid != entitybus!!.eid) return false
        return if (if (enumber != null) enumber != entitybus.enumber else entitybus.enumber != null) false else true

    }

    override fun hashCode(): Int {
        var result = eid
        result = 31 * result + if (enumber != null) enumber!!.hashCode() else 0
        return result
    }


}
